internal/impl/kube.go

```diff
@@ -600,34 +625,28 @@ func buildService(rs *replicaSetInfo, dep *protos.Deployment) (*corev1.Service,
 // it has to be reachable from the outside; for internal listeners, we generate
 // a ClusterIP service, reachable only from internal Service Weaver services.
 func buildListenerService(rs *replicaSetInfo, lis *ReplicaSetConfig_Listener, dep *protos.Deployment) (*corev1.Service, error) {
-	appName := name{dep.App.Name, rs.name, dep.Id[:8]}.DNSLabel()
-	lisName := name{dep.App.Name, "lis", rs.name, dep.Id[:8]}.DNSLabel()
+	// Unique name that persists across app versions.
+	globalLisName := name{dep.App.Name, "lis", rs.name}.DNSLabel()
+
 	var serviceType string
+	labels := map[string]string{}
 	if lis.IsPublic {
```

Comment:
I believe that if the listener is non-public, we should do the same thing, i.e., use a non-versioned name.

The way you have it in `buildAutoscaler` should work here I believe.

Reply:
Great point. Done